src/shims/fs.rs

```diff
@@ -375,6 +375,19 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 fh.insert_fd_with_min_fd(FileHandle { file: duplicated, writable }, start)
             });
             this.try_unwrap_io_result(fd_result)
+        } else if this.tcx.sess.target.target.target_os == "macos"
+            && cmd == this.eval_libc_i32("F_FULLFSYNC")?
+        {
+            // On macOS, fsync does not wait for the underlying disk to finish writing, while this
+            // F_FULLFSYNC operation does. The standard library uses F_FULLFSYNC for both
```

Comment:
```suggestion
            // F_FULLFSYNC implementation does. IOW, we conservatively implement this stronger than necessary.
            // The standard library uses F_FULLFSYNC for both
```

Reply:
Ah no that's not right either. The comment should be in `fsync`, which is the place where we are being conservative.

Reply:
Got it, added a comment there to that effect